xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -491,19 +492,23 @@ ManagedChannel createChannel(List<ServerInfo> servers) {
         ServerInfo serverInfo = servers.get(0);
         String serverUri = serverInfo.getServerUri();
         List<ChannelCreds> channelCredsList = serverInfo.getChannelCredentials();
-        ManagedChannel ch = null;
+        ManagedChannelBuilder<?> channelBuilder = null;
         // Use the first supported channel credentials configuration.
         // Currently, only "google_default" is supported.
         for (ChannelCreds creds : channelCredsList) {
           if (creds.getType().equals("google_default")) {
-            ch = GoogleDefaultChannelBuilder.forTarget(serverUri).build();
+            channelBuilder = GoogleDefaultChannelBuilder.forTarget(serverUri);
             break;
           }
         }
-        if (ch == null) {
-          ch = ManagedChannelBuilder.forTarget(serverUri).build();
+        if (channelBuilder == null) {
+          channelBuilder = ManagedChannelBuilder.forTarget(serverUri);
         }
-        return ch;
+
+        return channelBuilder
+            .keepAliveTime(5, TimeUnit.MINUTES)
+            .keepAliveTimeout(20, TimeUnit.SECONDS)
```

Comment:
I was thinking we'd just leave Timeout as its default. Yes, the default is 20 seconds, but no need to specify that here.

Reply:
Our internal prod channel builder has a different timeout. What's the specification? Use channel's default keepalive timeout or use 20 seconds?

Reply:
Each language handles internal vs external differently. I think Java and C have some options here, since the keepalive setting is in globalish state. I think Go may not have many options, since it probably wouldn't get the internal configuration naturally.

I don't think 10 seconds vs 20 seconds makes much of a difference internally, so in some ways I don't really care. More importantly we will be overriding the default internal keepalive time of 20 seconds and making it 5 minutes, which has a much bigger effect.

Options that come to mind:
 1. Don't care. We said 5 minutes will be fine, use 5 minutes. Could be surprising since it has different configuration from everything else
 2. Add configuration. Internally we can set different values in the bootstrap file. Could be surprising since it has different configuration from everything else
 3. Figure out a hack to use the default keepalive internally but set it to 5 minutes externally.
 4. Change the default keepalive for channels to something reasonable, and then not specify anything here. Too risky short-term. We know we already want to change the default keepalive for channels

Right now I think I'm favoring (1), with potential to do (2) and/or (4) in the future. We would leave timeout to be its system-specified default (so not specify it here).

CC @markdroth